### PR TITLE
Profile backgrounds

### LINKED ISSUE #120: Generate Dynamic Profile Background Image per User

### PR DESCRIPTION
**Pull request by SheepTester**
# Info

Closes #120 

generates a pie chart based on the communities of "recently attended" on the profile, then blurs the shapes individually

## Changes

- increased banner size

# Type of Change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) 

# Testing

I have tested that my changes fully resolve the linked issue ...

- [x] locally on Desktop.
- [ ] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have added new Cypress tests that are passing.

# Checklist

- [x] I have performed a self-review of my own code.
- [x] I have followed the style guidelines of this project.
- [x] I have documented any new functions in `/src/lib/*` and commented hard to understand areas
      anywhere else.
- [x] My changes produce no new warnings.

# Screenshots

https://github.com/acmucsd/membership-portal-ui-v2/assets/22133785/3c2f9be3-6a0b-479e-b168-7aae59fa3d35

### CODE REVIEW

File: src/components/profile/Banner/index.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  const communities: Record<Community, number> = {
+    [Community.HACK]: 0,
+    [Community.AI]: 0,
+    [Community.CYBER]: 0,
+    [Community.DESIGN]: 0,
+    [Community.GENERAL]: 0,
+  };
+  let total = 0;
+
+  recentAttendances.forEach(({ event: { committee } }) => {
```
Comment (farisashai) [inline src/components/profile/Banner/index.tsx:48]: right now if recentAttendances is empty the background is just missing - maybe just use a completely random color arrangement from the seeded uuid instead?

Comment (SheepTester) [reply src/components/profile/Banner/index.tsx:48]: hmm, or maybe it should be blank? like when the apple card colors reset. so it'll be like each attended event paints the background with a color

Comment (alexzhang1618) [reply src/components/profile/Banner/index.tsx:48]: > hmm, or maybe it should be blank?

I like this idea, maybe we have the blank base be a different color from the card though

Comment (SheepTester) [reply src/components/profile/Banner/index.tsx:48]: I could also just outright remove the banner if there are no attendances

**Review by farisashai — APPROVED** [review]
looks good to merge whenever you decide on missing backgrounds

**Comment by SheepTester** [comment]
for now we'll just hide banners on profiles without any attendance history. we can decide on a better option in the future

**Pull request merged**